### PR TITLE
populate retry interval for mercury requests and handle 206 response code

### PR DESCRIPTION
**Pull request by FelixFan1992**
This PR targets AUTO-6936
- It sets retryable to true for response code 502 in both v02 and v03
- Additionally for mercury v03 it handles code 206 as retryable (partial reports)
- It populates a retry interval for retryable failed mercury requests.
- For the first 5 retryable failed pipeline runs caused by mercury, it will populate a 1s retry interval
- For the last 5 retryable failed pipeline runs caused by mercury, it will populate a 5s retry interval
- After that, it will set retryable to false

This PR also targets AUTO-5044
- It handles potential 206 response code from mercury server
- This response code means the mercury request is valid but mercury server has not generated reports for some of the feeds
- It treats 206 as a retryable response code

This PR also targets AUTO-7090
- It handles 502, 503, 504 response codes from mercury server and treats them as retryable

This PR also targets AUTO-6956
- It reduces mercury permission cache period from 20 mins to 10 mins

### CODE REVIEW

File: core/services/ocr2/plugins/ocr2keeper/evm21/streams_lookup.go [hunk 1]
```diff
@@ -299,29 +300,30 @@ func (r *EvmRegistry) checkCallback(ctx context.Context, values [][]byte, lookup
...
-		return encoding.NoPipelineError, encoding.UpkeepFailureReasonInvalidRevertDataInput, [][]byte{}, false, fmt.Errorf("invalid revert data input: feed param key %s, time param key %s, feeds %s", sl.feedParamKey, sl.timeParamKey, sl.feeds)
+		return encoding.NoPipelineError, encoding.UpkeepFailureReasonInvalidRevertDataInput, [][]byte{}, false, 0 * time.Second, fmt.Errorf("invalid revert data input: feed param key %s, time param key %s, feeds %s", sl.feedParamKey, sl.timeParamKey, sl.feeds)
 	}
 	if sl.feedParamKey == feedIdHex && sl.timeParamKey == blockNumber {
 		// only mercury v0.2
 		for i := range sl.feeds {
 			go r.singleFeedRequest(ctx, ch, i, sl, lggr)
 		}
-	} else if sl.feedParamKey == feedIDs && sl.timeParamKey == timestamp {
+	} else if sl.feedParamKey == feedIDs {
```
Comment (FelixFan1992) [inline core/services/ocr2/plugins/ocr2keeper/evm21/streams_lookup.go:315]: should i remove this?

File: core/services/ocr2/plugins/ocr2keeper/evm21/streams_lookup.go [hunk 2]
```diff
@@ -132,13 +132,13 @@ func (r *EvmRegistry) streamsLookup(ctx context.Context, checkResults []ocr2keep
 			}
 
 			if !allowed {
-				lggr.Warnf("at block %d upkeep %s NOT allowed to query Mercury server", block, upkeepId)
+				lggr.Debugf("at block %d upkeep %s NOT allowed to query Mercury server", block, upkeepId)
 				checkResults[i].IneligibilityReason = uint8(encoding.UpkeepFailureReasonMercuryAccessNotAllowed)
 				continue
 			}
-		} else if l.feedParamKey != feedIDs || l.timeParamKey != timestamp {
+		} else if l.feedParamKey != feedIDs {
```
Comment (FelixFan1992) [inline core/services/ocr2/plugins/ocr2keeper/evm21/streams_lookup.go:139]: same here

File: core/services/ocr2/plugins/ocr2keeper/evm21/streams_lookup.go (hunk 1 above)
Comment (FelixFan1992) [reply core/services/ocr2/plugins/ocr2keeper/evm21/streams_lookup.go:315]: removed for now

File: core/services/ocr2/plugins/ocr2keeper/evm21/streams_lookup.go [hunk 3]
```diff
@@ -132,13 +132,13 @@ func (r *EvmRegistry) streamsLookup(ctx context.Context, checkResults []ocr2keep
...
 			// if mercury version cannot be determined, set failure reason
-			lggr.Warnf("at block %d upkeep %s NOT allowed to query Mercury server", block, upkeepId)
+			lggr.Debugf("at block %d upkeep %s NOT allowed to query Mercury server", block, upkeepId)
```
Comment (infiloop2) [inline core/services/ocr2/plugins/ocr2keeper/evm21/streams_lookup.go:141]: These look unrelated to the change

File: core/services/ocr2/plugins/ocr2keeper/evm21/streams_lookup.go (hunk 1 above)
Comment (infiloop2) [inline core/services/ocr2/plugins/ocr2keeper/evm21/streams_lookup.go:315]: unrelated

File: core/services/ocr2/plugins/ocr2keeper/evm21/streams_lookup.go (hunk 2 above)
Comment (infiloop2) [inline core/services/ocr2/plugins/ocr2keeper/evm21/streams_lookup.go:139]: unrelated change

File: core/services/ocr2/plugins/ocr2keeper/evm21/streams_lookup.go [hunk 4]
```diff
@@ -520,16 +525,16 @@ func (r *EvmRegistry) multiFeedsRequest(ctx context.Context, ch chan<- MercuryDa
...
 				retryable = false
 				state = encoding.InvalidMercuryRequest
-				return fmt.Errorf("at timestamp %s upkeep %s received status code %d from mercury v0.3, most likely this is caused by invalid format of timestamp", sl.time.String(), sl.upkeepId.String(), resp.StatusCode)
-			} else if resp.StatusCode == http.StatusInternalServerError {
+				return fmt.Errorf("at timestamp %s upkeep %s received status code %d from mercury v0.3 with message: %s", sl.time.String(), sl.upkeepId.String(), resp.StatusCode, string(body))
+			} else if resp.StatusCode == http.StatusInternalServerError || resp.StatusCode == http.StatusBadGateway || resp.StatusCode == http.StatusServiceUnavailable || resp.StatusCode == http.StatusGatewayTimeout {
 				retryable = true
 				state = encoding.MercuryFlakyFailure
-				return fmt.Errorf("%d", http.StatusInternalServerError)
-			} else if resp.StatusCode == 420 {
```
Comment (infiloop2) [inline core/services/ocr2/plugins/ocr2keeper/evm21/streams_lookup.go:528]: we no longer care about 420? is this part of this PR change?

Comment (infiloop2) [reply core/services/ocr2/plugins/ocr2keeper/evm21/streams_lookup.go:528]: discussed offline, this is intended
